app/lib/features/chat_ng/widgets/events/reply_original_event.dart

```diff
@@ -0,0 +1,168 @@
+  final String roomId;
+  final RoomMessage msg;
+  const ReplyOriginalEvent({
+    super.key,
+    required this.roomId,
+    required this.msg,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    // reply is allowed for only EventItem not VirtualItem
+    // user should be able to get original event as RoomMessage
+    RoomEventItem replyEventItem =
+        msg.eventItem().expect('room msg should have event item');
+    final messageId = msg.uniqueId();
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Consumer(
+          builder: (context, ref, child) => replyProfileBuilder(
+            context,
+            ref,
+            roomId,
+            replyEventItem,
+          ),
+        ),
+        // const SizedBox(height: 10),
+        Padding(
+          padding: const EdgeInsets.fromLTRB(10, 10, 10, 0),
+          child: OriginalEventItem(
+            roomId: roomId,
+            messageId: messageId,
+            item: replyEventItem,
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget replyProfileBuilder(
+    BuildContext context,
+    WidgetRef ref,
+    String roomId,
+    RoomEventItem item,
+  ) {
+    final replyProfile = ref.watch(
+      memberAvatarInfoProvider((userId: item.sender(), roomId: roomId)),
+    );
+
+    return Row(
+      children: [
+        ActerAvatar(
+          options: AvatarOptions.DM(
+            replyProfile,
+            size: 12,
+          ),
+        ),
+        const SizedBox(width: 5),
+        Text(
+          replyProfile.displayName ?? '',
+          style: Theme.of(context)
+              .textTheme
+              .bodySmall
+              ?.copyWith(color: Theme.of(context).colorScheme.tertiary),
+        ),
+      ],
+    );
+  }
+}
+
+class OriginalEventItem extends ConsumerWidget {
+  final String roomId;
+  final String messageId;
+  final RoomEventItem item;
+  const OriginalEventItem({
+    super.key,
+    required this.roomId,
+    required this.messageId,
+    required this.item,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final myId = ref.watch(myUserIdStrProvider);
+    final isUser = myId == item.sender();
+    final eventType = item.eventType();
+    return switch (eventType) {
+      // handle message inner types separately
+      'm.room.message' => buildReplyMsgEventItem(
+          roomId,
+          messageId,
+          item,
+          isUser,
+        ),
+      'm.room.redaction' => RedactedMessageWidget(),
+      'm.room.encrypted' => EncryptedMessageWidget(),
+      _ => _buildUnsupportedMessage(eventType),
+    };
+  }
+
+  Widget buildReplyMsgEventItem(
+    String roomId,
+    String messageId,
+    RoomEventItem item,
+    bool isUser,
+  ) {
+    final msgType = item.msgType();
+    final content = item.msgContent();
+    final metadata = MessageMetadata(
+      roomId: roomId,
+      messageId: messageId,
+      isUser: isUser,
+      isReply: true,
```

Comment:
this is such an important flag which indicating _how_ the redering is supposed to happen, yet it is bundled into a `metadata`-object that is about the messages. The messages indicated here is _not_ a reply, it is a message to be replied to - and that flag only matters for the rendering, this isn't _metadata_ but _rendering information_. We should really differentiate between the concerns. This _flag_ should really be a `TextMessagesEvent.noBubble`-factory I think...